go/dexop/dex/client.go

```diff
@@ -0,0 +1,146 @@
+package dex
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"os"
+
+	dexapi "github.com/dexidp/dex/api/v2"
+	dexv1alpha1 "github.com/rackerlabs/understack/go/dexop/api/v1alpha1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type DexManager struct {
+	Client dexapi.DexClient
+	Issuer string
+}
+
+// Creates new Oauth2 client in Dex
+func (d *DexManager) CreateOauth2Client(clientSpec *dexv1alpha1.Client) (*dexapi.CreateClientResp, error) {
+	request := &dexapi.CreateClientReq{
```

Comment:
nitpic: not sure if we do need to check if `clientSpec` is not nill here.

Reply:
it shouldn't be, but input validation is always good so I added it anyway